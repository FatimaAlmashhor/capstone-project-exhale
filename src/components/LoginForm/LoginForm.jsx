import React, { useState } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useTranslation } from 'react-i18next';
import fire from '../../firebase';

export default function LoginForm() {
  const { t } = useTranslation();
  const ReviewError = Yup.object().shape({
    email: Yup.string().email().required(t('EmailRequired')),
    password: Yup.string().required(t('PassRequired')),
  });
  const [errMessage, setErrMessage] = useState('');
  return (
    <Formik
      initialValues={{ email: '', password: '' }}
      validationSchema={ReviewError}
      onSubmit={(values, { setSubmitting }) => {
        setTimeout(() => {
          fire
            .auth()
            .signInWithEmailAndPassword(values.email, values.password)
            .catch((err) => {
              setErrMessage(err.message);
            });
          setSubmitting(false);
        }, 500);
      }}
    >
      {(formik) => (
        <div className="w-full max-w-xs mx-auto">
          <form
            onSubmit={formik.handleSubmit}
            className="pt-6 pb-3 mb-4 bg-white rounded"
          >
            <input
              className="w-full p-3 leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
              id="email"
              name="email"
              type="text"
              placeholder={t('yourEmail')}
              onChange={formik.handleChange}
            />
            {formik.errors.email && (
              <div className="text-sm font-bold text-red-500">
                {formik.errors.email}
              </div>
            )}

            <input
              className="w-full p-3 mt-5 leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
              id="password"
              name="password"
              type="password"
              placeholder={t('yourPassword')}
              onChange={formik.handleChange}
            />
            {formik.errors.password && (
              <div className="text-sm font-bold text-red-500">
                {formik.errors.password}
              </div>
            )}
            <div className="flex items-center justify-between mb-6">
              <button
                className="inline-block w-full px-4 py-3 mt-5 text-white bg-blue-800 rounded shadow-lg hover:bg-blue-600 focus:bg-blue-700"
                type="submit"
              >
                {t('Login')}
              </button>
            </div>
            <div className="text-lg text-red-500">{errMessage}</div>
          </form>
        </div>
      )}
    </Formik>
  );
}
