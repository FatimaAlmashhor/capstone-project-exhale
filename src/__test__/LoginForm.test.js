import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as  Router} from 'react-router-dom'
import LoginForm from '../components/LoginForm';

it('renders correctly', () => {
  const tree = renderer.create(<Router><LoginForm  /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
