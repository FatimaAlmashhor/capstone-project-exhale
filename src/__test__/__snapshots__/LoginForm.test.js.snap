// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full max-w-xs mx-auto"
>
  <form
    className="pt-6 pb-3 mb-4 bg-white rounded"
    onSubmit={[Function]}
  >
    <input
      className="w-full p-3 leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="yourEmail"
      type="text"
    />
    <input
      className="w-full p-3 mt-5 leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="yourPassword"
      type="password"
    />
    <div
      className="flex items-center justify-between mb-6"
    >
      <button
        className="inline-block w-full px-4 py-3 mt-5 text-white bg-blue-800 rounded shadow-lg hover:bg-blue-600 focus:bg-blue-700"
        type="submit"
      >
        Login
      </button>
    </div>
    <div
      className="text-lg text-red-500"
    >
      
    </div>
  </form>
</div>
`;
